Storybook v8 stories batch 7

misama-ct opened the pull request:
## Background
In an effort to switch from Storybook v5 to Storybook v8, the existing stories had to be migrated to a new storybook-format and converted to typescript. 

This pull-request is [part of a batch](https://github.com/commercetools/ui-kit/pull/2875). It contains the typescript-equivalents of existing storybook v5 stories and aims to replicate the same functionality / value. 

## Goal
Feature parity between v8 and v5 storybook. After merging all batches, a product developer who looked at the storybook v5 version yesterday, should be able to look at the v8 version tomorrow and be as productive (or more productive) as before. 

## Review instructions
A thorough **code review** is not required yet. The code was either copied from the existing js-equivalent or quickly written from scratch to replicate what was already there to fit the new story-format. It is expected that those stories contain typescript errors, introduced by the conversion or previously existing. (Another pull-request will take care of those issues at a later stage).

What is expect is a **story review**:

You will need to compare the v5 with the v8 storybook (ideally side-by-side) and make sure that every component in this pull-request has:

- [ ] a readme-file in the same or a parent-folder
- [ ] a props-table displaying available props (and appropriate inputs for them)
- [ ] one or more stories that showcase the same or more functionality as the v5 equivalent

You will find the links to the different storybooks below. 


> If something is missing or irritating, add a comment to the source file here in the pull-request to start a conversation.

FilPob commented:
Stumbled across the content notification story. Per default the Type is not set. That leads to the notification being shown with a style that is not intended 
<img width="1073" alt="image" src="https://github.com/user-attachments/assets/f73c5e73-ecaf-4f1b-a5b3-62dc20d6ee78">

ddouglasz commented:
**PrimaryActionDropDown**
I am missing the disabled and label properties for interacting with each option

<img width="658" alt="image" src="https://github.com/user-attachments/assets/4a484e82-2345-46cb-b901-345699dff5f2">

ddouglasz commented:
**WarningMessage**
clicking the intlMessage control breaks the story

<img width="1604" alt="image" src="https://github.com/user-attachments/assets/8c33e23e-e8b0-472a-a18f-9692930f55ff">

ddouglasz commented:
**ContentNotification**

1. Clicking `intlMessage` control also breaks the story
2. We are missing the conrtol for `onRemove`

ddouglasz commented:
**Stamp**
Clicking the children control breaks the Story
![2024-08-14 14 10 09](https://github.com/user-attachments/assets/8726940d-afe6-4e3b-8316-919cc71b1ecd)

ddouglasz commented:
**ViewSwitcher**
Removing any of the items from the `children` control breaks the Story
![2024-08-14 14 23 18](https://github.com/user-attachments/assets/b7092810-89d4-4ea2-9aef-393b0532c3a2)

misama-ct commented:
> Stumbled across the content notification story. Per default the Type is not set. That leads to the notification being shown with a style that is not intended

@FilPob 

✅ Fixed

misama-ct commented:
> **PrimaryActionDropDown** I am missing the disabled and label properties for interacting with each option
> 

@ddouglasz 
I didn't even attempt to replicate this. Even if it's somehow possible in sb v8 with some hacks, I don't think it adds value. The Basic example contains a disabled Item, the basic examples' code demonstrates every `Option` prop and the tab-bar below the example allows to show the prop-table for the `Option` component as well. 

![image](https://github.com/user-attachments/assets/d5839cac-5132-4105-8c82-dd298ace8310)

misama-ct commented:
> **WarningMessage** clicking the intlMessage control breaks the story
> 
@ddouglasz 

✅ I deactivated the `intlMessage` control.

misama-ct commented:
> **Stamp**
> Clicking the children control breaks the Story

@ddouglasz 
✅  According to the `TStampProps` type, `children` is deprecated in favor of `label`. I added the `@deprecated` annotation to the `children` prop's  typescript description and removed the `control` for `children` from the table.

misama-ct commented:
> **ContentNotification**
> 
> 1. Clicking `intlMessage` control also breaks the story
> 2. We are missing the conrtol for `onRemove`

@ddouglasz 
1. ✅ Fixed (I removed the control-input from the table)
2. There is the extra story _Dismissable Notification_ to demonstrate the `onRemove` prop (= if onRemove is defined, show the closing button)

misama-ct commented:
> **ViewSwitcher**
> Removing any of the items from the `children` control breaks the Story

@ddouglasz 

I removed the `children` control from the props-table since the value needs to be a `ReacNode` and this can not be handled by any regular sb control-input. I also removed the `defaultSelected` control, because its' effect can not be demonstrated after the component has been mounted already.

ddouglasz reviewed (approved):
Thank you for the follow up! 🙌🏽

ByronDWall commented on packages/components/messages/src/error-message/error-message.stories.tsx:
```diff
@@ -0,0 +1,16 @@
+
+const meta: Meta<typeof ErrorMessage> = {
+  title: 'components/Messages/ErrorMessage',
+  component: ErrorMessage,
```
Like in `WarningMessage`, setting an `intlMessage` breaks the example component.
```suggestion
  component: ErrorMessage,
    argTypes: {
    intlMessage: { control: { disable: true } },
  },
```

ByronDWall reviewed (approved):
One small nit, but Im going to go ahead and approve.  The spacings in particular are a big improvement, thanks!
